src/Doctrine/Common/State/PersistProcessor.php

```diff
@@ -52,7 +52,7 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             \assert(method_exists($manager, 'getReference'));
             $newData = $data;
             $identifiers = array_reverse($uriVariables);
-            $links = $this->getLinks($class, $operation, $context);
+            $links = $this->getLinks($context['resource_class'] ?? $class, $operation, $context);
```

Comment:
can you add a functional test with your stateOption? couldn't we get `stateOptions->entityClass` directly instead of using this context value?

Reply:
@soyuka thanks for your reply. The `stateOptions->entityClass` could be null if not applied to an API Resource, thats why i opted for context usage.
I've just added a Functional Tests to the Branch.

Reply:
Question is, do we need the changes to the Persist Process generally?
I'm sadly not that deep into API Platform to determine it right now.

For me, it was important to start working with the right Pattern, which always concludes the separation of Entities from View / Logic, but relying on existing Logic as much as Possible. Therefore i opted for the DTO Approach using stateOptions to still use the Doctrine Handler.

As it seems not a lot of people / projects are using the DTO Approach, specially with GraphQL, i would guess the Bug need some further investigation by People with more knowledge into API Platform.

Maybe someone can have a deeper look into this PR and Issue.

Reply:
IMHO these lines are missing here: 

https://github.com/api-platform/core/blob/a50e2d03928c2e330187772f3d3062696733062d/src/Doctrine/Orm/State/ItemProvider.php#L51-L55


Then the `entityClass` will be correct